### PR TITLE
API functional tests: interaction with databases

### PR DESCRIPTION
**Pull request by arothuis**
See my (MKodde) commits for more information about the individual changes that make up 
this PR. The changeset is quite large.

[112497975](https://www.pivotaltracker.com/story/show/112497975)

Based on: #409
* [x] That PR should be merged first. This PR shall be updated accordingly.

First thing:
* [x] Resolve conflicts and rebase on master

Todo:
* [x] Correct and merge PR #409
* [x] Add more extensive functional tests for Connection API
* [x] Provision test database in Travis
* [x] Provision test database in OpenConext-deploy (dev)
* [x] Fix build for Behat functional tests (related to the used [MetadataRepository](https://github.com/OpenConext/OpenConext-engineblock/pull/413/commits/c14de574cbc97d37f06ff63fc44ec27d5848c2d2)
* [x] Clean up loose fixtures to fixture classes

### CODE REVIEW

File: library/EngineBlock/Application/DiContainer.php [hunk 1]
```diff
@@ -134,8 +136,23 @@ public function getAuthenticationLoopGuard()
         return $this->container->get('engineblock.authentication.authentication_loop_guard');
     }
 
-    protected function registerMetadataRepository()
+    protected function registerMetadataRepository($env)
     {
+        if ($env === 'test') {
```
Comment (arothuis) [inline library/EngineBlock/Application/DiContainer.php]: This possibly messes things up for our Behat functional tests. 
Perhaps only do this if `$functionalTesting !== true`?

Also, a comment should be added to explain why this is necessary.

Comment (arothuis) [inline library/EngineBlock/Application/DiContainer.php]: This should be done in the test-specific `TestDiContainer`.

File: tests/functional/OpenConext/EngineBlockBundle/Controller/Api/AttributeReleasePolicyControllerTest.php [hunk 2]
```diff
@@ -153,4 +440,66 @@ public function invalidJsonPayloadProvider()
         ];
     }
 
+    private function createServiceProviderWithArp($entityId, AttributeReleasePolicy $attributeReleasePolicy)
+    {
+        return new ServiceProvider(
```
Comment (jorissteyn) [inline tests/functional/OpenConext/EngineBlockBundle/Controller/Api/AttributeReleasePolicyControllerTest.php]: I generally too prefer constructor injection and immutability over a bunch of public properties. But I dislike this change, what's wrong with how it used to work? All places in EB use it like this:

```
$sp = new ServiceProvider($entityId);
$sp->attributeReleasePolicy = $attributeReleasePolicy;
```

Same goes for identityProvider.

Comment (jorissteyn) [reply tests/functional/OpenConext/EngineBlockBundle/Controller/Api/AttributeReleasePolicyControllerTest.php]: See ServiceRegistryFixture::register* methods for more inspiration!

Comment (MKodde) [reply tests/functional/OpenConext/EngineBlockBundle/Controller/Api/AttributeReleasePolicyControllerTest.php]: Good point. I've addressed this in commit 6823bcd

**Pull request merged**